### PR TITLE
BaSP-TG-4: Projects Page - List and Delete

### PR DESCRIPTION
**Pull request by DamianPalavecino**
- Show table with projects list - Columns: Project name, Start date, Description, Client, Employees, Status, Actions.
- Add delete row functionality
- Add actions icon (delete and edit) for each row
- Create modal for warning message when deleting a project
- Create modal for Success message when project is deleted
- Add "Add new Project" button and redirect to add form
- Add a button 'See Employees'. This redirects to a list of employees per project

### CODE REVIEW

File: src/Components/Projects/index.jsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        </thead>
+        <tbody className={styles.tbody}>
+          <Modal
+            project={modalObject}
+            handleDelete={handleDelete}
+            type={modalType}
+            showModal={showModal}
+            closeModal={closeModal}
+          />
+          {projects === undefined && <p>There are no projects</p>}
```
Comment (Juanoauil) [inline src/Components/Projects/index.jsx:67]: you can use {`projects.length > 0 ? projects.map(() => ... : <p> There are no projects </p>`}

**Review by Juanoauil — CHANGES_REQUESTED** [review]

**Review by martumoser — CHANGES_REQUESTED** [review]

File: src/Components/Projects/index.jsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
+import { useEffect, useState } from 'react';
+import Modal from './modal';
+import styles from './projects.module.css';
+
+const Projects = () => {
+  const [projects, saveProjects] = useState([]);
+  const [showModal, setShowModal] = useState(false);
+  const [modalType, changeModalType] = useState('');
+  const [modalObject, setModalObject] = useState();
```
[the remark this answers is not in the document]
Comment (DamianPalavecino) [reply src/Components/Projects/index.jsx:9]: I don't have two modals. modalType is where I set what type of modal do I want to open (success, delete, employees) and modalObject is where I save a specific project object to delete it or show its employees by using the modal

**Review by martumoser — DISMISSED** [review]

**Review by PilarFernandezC — APPROVED** [review]

File: src/Components/Projects/index.jsx (hunk 1 above)
Comment (Juanoauil) [reply src/Components/Projects/index.jsx]: not well applied

**Review by Juanoauil — CHANGES_REQUESTED** [review]

**Review by Juanoauil — APPROVED** [review]

**Review by Juanoauil — APPROVED** [review]

**Review by mara666 — APPROVED** [review]
Good job Dami!

**Pull request merged**